src/Testing/CodeHacking/CodeHackerTestHook.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * CodeHackerTestHook class file.
+ *
+ * @package WooCommerce/Testing
+ */
+
+// phpcs:disable Squiz.Commenting.FunctionComment.Missing, PHPCompatibility.FunctionDeclarations
+
+namespace Automattic\WooCommerce\Testing\CodeHacking;
+
+use PHPUnit\Runner\BeforeTestHook;
+use PHPUnit\Runner\AfterTestHook;
+use PHPUnit\Util\Test;
+use ReflectionClass;
+use ReflectionMethod;
+use Exception;
+
+/**
+ * Helper to use the CodeHacker class in PHPUnit.
+ *
+ * How to use:
+ *
+ * 1. Add this to phpunit.xml:
+*
+ *    <extensions>
+ *      <extension class="CodeHackerTestHook" />
+ *    </extensions>
+ *
+ * 2. Add the following to the test classes:
+ *
+ *    use Automattic\WooCommerce\Testing\CodeHacking\CodeHacker;
+ *    use Automattic\WooCommerce\Testing\CodeHacking\Hacks\...
+ *
+ *    public static function before_all($method_name) {
```

Comment:
Why don't we use `setup()` instead of having to implement `before_all`?

Reply:
I see the explanation for the above comment in the readme, but why can; 't we call setUp from where ever we call `before_all`?

Reply:
If we call `setUp` here then it will be called again by PHPUnit itself later, and this could cause problems (or at least slow down the execution of the test suite).

Reply:
ok, makes sense